pandas/core/arrays/base.py

```diff
@@ -333,6 +335,18 @@ def __iter__(self):
         for i in range(len(self)):
             yield self[i]
 
+    def __eq__(self, other: Any) -> Union[np.ndarray, "ExtensionArray"]:
+        """
+        Return for `self == other` (element-wise equality).
```

Comment:
I think we should have some guidance here that if `other` is a Series then you should return NotImplemented.

Reply:
could use ops.common.unpack_zerodim_and_defer

Reply:
> could use ops.common.unpack_zerodim_and_defer

This method is to be implemented by EA authors, so those can't use that helper (unless we expose somewhere a public version of this). 

(we could of course use that for our own EAs, but this PR is not changing any existing `__eq__` implementation at the moment)

Reply:
> I think we should have some guidance here that if other is a Series then you should return NotImplemented.

Added a comment about that